grakn-graph/src/main/java/ai/grakn/graph/internal/InstanceImpl.java

```diff
@@ -59,7 +59,17 @@
  */
 abstract class InstanceImpl<T extends Instance, V extends Type> extends ConceptImpl<T> implements Instance {
     private ComponentCache<TypeLabel> cachedInternalType = new ComponentCache<>(() -> TypeLabel.of(getProperty(Schema.ConceptProperty.TYPE)));
-    private ComponentCache<V> cachedType = new ComponentCache<>(() -> this.<V>getOutgoingNeighbours(Schema.EdgeLabel.ISA).findFirst().orElse(null));
+
+    private ComponentCache<V> cachedType = new ComponentCache<>(() -> {
+        ConceptImpl<?> currentType = (ConceptImpl) this.getOutgoingNeighbours(Schema.EdgeLabel.ISA).findFirst().orElse(null);
+
+        while (currentType.isShard()){
```

Comment:
Does this have to be a loop? Given we only have one depth of shards it seems unnecessary. If this is just a matter of preparing for when we do have a tree of shards, then that's alright - although I'd face that when it comes.